axelrod/strategies/axelrod_second.py

```diff
@@ -975,3 +975,297 @@ def strategy(self, opponent: Player) -> Action:
+
+        self.chi_squared = round(chi_squared, 3) # For testing
+
+        if chi_squared > 3:
+            return False
+        return True
+
+    def detect_streak(self, last_move):
+        """
+        Return if and only if the opponent's last twenty moves are defects.
+        """
+
+        if last_move == D:
+            self.defect_streak += 1
+        else:
+            self.defect_streak = 0
+        if self.defect_streak >= 20:
+            return True
+        return False
+
+    def detect_parity_streak(self, last_move):
+        self.parity_bit = 1 - self.parity_bit # Flip bit
+        if last_move == D:
+            self.parity_streak[self.parity_bit] += 1
+        else:
+            self.parity_streak[self.parity_bit] = 0
+        if self.parity_streak[self.parity_bit] >= self.parity_limit:
+            return True
+
+    def strategy(self, opponent: Player) -> Action:
+        turn = len(self.history) + 1
+
+        if turn == 1:
+            return C
+
+        if self.mode == "Defect":
+            if opponent.history[-1] == D:
+                self.exit_defect_meter += 1
+            else:
+                self.exit_defect_meter -= 3
+            if self.exit_defect_meter >= 11:
+                self.mode = "Normal"
+                self.was_defective = True
+                self.more_coop = 2
+                return self.try_return(to_return=C, lower_flags=False)
+
+            return self.try_return(D)
+
+
+        # If not Defect mode, proceed to update history and check for random,
+        # check if burned, and check if opponent's fairweather.
+
+        # History only gets updated outside of Defect mode.
+        if turn > 2:
+            if opponent.history[-1] == D:
+                self.recorded_defects += 1
+            opp_col = 1 if opponent.history[-1] == D else 0
+            self.move_history[self.history_row, opp_col] += 1
+
+        # Detect random
+        if turn % 15 == 0 and turn > 15 and not self.was_defective:
+            if self.detect_random(turn):
+                self.mode = "Defect"
+                return self.try_return(D, lower_flags=False) # Lower_flags not used here.
+
+        # history_row only gets updated if not in Defect mode AND not entering
+        # Defect mode.
+        self.history_row = 1 if opponent.history[-1] == D else 0
+        if self.history[-1] == D:
+            self.history_row += 2
+
+        # If generous 2 turn ago and opponent defected last turn
+        if self.generous_n_turns_ago == 2 and opponent.history[-1] == D:
+            self.burned = True
+
+        if turn == 38 and opponent.history[-1] == D and opponent.cooperations == 36:
+            self.mode = "Fair-weather"
+            return self.try_return(to_return=C, lower_flags=False)
+
+
+        if self.mode == "Fair-weather":
+            if opponent.history[-1] == D:
+                self.mode = "Normal" # Post-Defect is not possible
+                #Continue below
```

Comment:
Remove this comment

Reply:
I'm trying to express here that this is the only place in the code where, following a mode-switch, we don't immediately return a value.  Instead we actually treat this turn as a "Normal" mode code.  I'll take your advice if you think it's really not worth mentioning.  (Or if I should say more clearly.)

Reply:
I don't feel strongly about it but it's clear that the code continues to run IMO